### PR TITLE
ATLEDGE-429 CurieSoftwareSerial fixes

### PR DESCRIPTION
**Pull request by bigdinotech**
-Creating a new CurieSoftwareSerial object no longer break functionality
of previous ones.
-Minor Cleanup

### CODE REVIEW

**Comment by bigdinotech** [comment]
@kmsywula @SidLeung ready for code review

File: libraries/CurieSoftwareSerial/SoftwareSerial.cpp [hunk 1]
```diff
@@ -46,7 +46,6 @@ volatile uint8_t SoftwareSerial::_receive_buffer_head = 0;
 //
 // Globals
 //
-uint8_t txPin;
```
Comment (kmsywula) [inline libraries/CurieSoftwareSerial/SoftwareSerial.cpp]: Why did you remove txPin?

File: libraries/CurieSoftwareSerial/SoftwareSerial.cpp [hunk 2]
```diff
@@ -95,8 +94,16 @@ bool SoftwareSerial::listen()
     bufferOverflow = false;
     _receive_buffer_head = _receive_buffer_tail = 0;
     active_object = this;
-
-    setRxIntMsk(true);
```
Comment (kmsywula) [inline libraries/CurieSoftwareSerial/SoftwareSerial.cpp]: Why removing setRxIntMsk?

File: libraries/CurieSoftwareSerial/SoftwareSerial.cpp [hunk 3]
```diff
@@ -230,9 +237,9 @@ SoftwareSerial::SoftwareSerial(uint32_t receivePin, uint32_t transmitPin, bool i
 {
   invertedLogic = inverse_logic;
   setTX(transmitPin);
-  txPin = transmitPin;
+  _transmitPin = transmitPin;
```
Comment (kmsywula) [inline libraries/CurieSoftwareSerial/SoftwareSerial.cpp]: Why do you use _transmitPin instead of txPin?

File: libraries/CurieSoftwareSerial/SoftwareSerial.h [hunk 4]
```diff
@@ -66,7 +67,6 @@ class SoftwareSerial : public Stream
   void tx_pin_write(uint32_t pin_state) __attribute__((__always_inline__));
   void setTX(uint8_t transmitPin);
   void setRX(uint8_t receivePin);
-  void setRxIntMsk(bool enable);
```
Comment (kmsywula) [inline libraries/CurieSoftwareSerial/SoftwareSerial.h]: Is this function no longer necessary? Maybe it should get removed from .cpp file as well?

File: libraries/CurieSoftwareSerial/SoftwareSerial.cpp (hunk 3 above)
Comment (bigdinotech) [inline libraries/CurieSoftwareSerial/SoftwareSerial.cpp]: txPin was originally a global variable in the cpp file 
i needed a variable that is an instance per each SoftwareSerial object.

File: libraries/CurieSoftwareSerial/SoftwareSerial.h (hunk 4 above)
Comment (bigdinotech) [inline libraries/CurieSoftwareSerial/SoftwareSerial.h]: I did remove it from the cpp file

File: libraries/CurieSoftwareSerial/SoftwareSerial.cpp (hunk 2 above)
Comment (bigdinotech) [inline libraries/CurieSoftwareSerial/SoftwareSerial.cpp]: removing since it is unused.

**Comment by kmsywula** [comment]
@calvinatintel code reviewed, please merge

**Comment by calvinatintel** [comment]
Cherry-picked
